File: app/src/main/java/jp/bellware/echo/setting/HtmlViewerActivity.java
package jp.bellware.echo.setting;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.MenuItem;
import android.view.View;
import android.webkit.WebView;

import jp.bellware.echo.R;

/**
 * HTML Viewer
 */
public class HtmlViewerActivity extends AppCompatActivity {

    public static final String EXTRA_URL = "url";
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_html_viewer);
        //ツールバーの設定
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        getSupportActionBar().setHomeButtonEnabled(true);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        //WebViewの設定
        WebView webView = (WebView) findViewById(R.id.web);
        webView.loadUrl(getIntent().getStringExtra(EXTRA_URL));
        webView.setOnLongClickListener(new View.OnLongClickListener() {
            @Override
            public boolean onLongClick(View v) {
                return true;
            }
        });
        webView.setHapticFeedbackEnabled(false);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if(item.getItemId() == android.R.id.home){
            finish();
            return true;
        }
        return false;
    }
}

File: app/src/main/java/jp/bellware/echo/setting/HtmlViewerActivity.kt
package jp.bellware.echo.setting

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.MenuItem
import kotlinx.android.synthetic.main.activity_html_viewer.*
import jp.bellware.echo.R

/**
 * HTML Viewer
 */
class HtmlViewerActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_html_viewer)
        //ツールバーの設定
        setSupportActionBar(toolbar)
        val lsab = supportActionBar
        lsab?.setHomeButtonEnabled(true)
        lsab?.setDisplayHomeAsUpEnabled(true)
        //WebViewの設定
        web.loadUrl(intent.getStringExtra(EXTRA_URL))
        web.setOnLongClickListener { true }
        web.isHapticFeedbackEnabled = false
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            finish()
            return true
        }
        return false
    }

    companion object {

        val EXTRA_URL = "url"
    }
}
